philip/src/components/Search.svelte

```diff
@@ -10,75 +11,88 @@
 
   // search user
   let username = "";
+  let profile = null;
+  let following = false;
   let outbox_collection = null;
-  let searched_profile = null;
 
   // search post
   let loadedPost = "";
   let postLink = "";
   let error = "";
 
   async function search(event) {
+    error = "";
+    profile = null;
     outbox_collection = null;
     let pair = username.split("@");
-    let profile_url = "https://" + pair[1] + "/@" + pair[0];
-    let collection;
+    if (pair.length !== 2) {
+      return (error = "Use this format: username@domain");
+    }
+    let profile_url = `${protocol}://${pair[1]}/@${pair[0]}`;
+
     if (pubgate_instance) {
-      const profile = await fetch(base_url + "/proxy", {
```

Comment:
what do you think about checking domain, and if it is same as `base_url`, do not use proxy and make direct request to API

Reply:
maybe factor out it in separate request util
like this one https://github.com/autogestion/pubgate-philip/blob/master/philip/src/utils/xhr.js

Reply:
1) good idea, will give it a shot
2) this was for #22? fixed in dbe1720